### PR TITLE
New feature: Integration of ogbl-vessel specific code

### PR DESCRIPTION
**Pull request by jqmcginnis**
Hi SEAL team,

we are currently moving towards the official release of [ogbl-vessel](https://ogb.stanford.edu/docs/linkprop/#ogbl-vessel) and would be very happy to have SEAL on the official leaderboard. Thus, I implemented some changes to the current code base to support the new graph. Please consider reviewing the changes and pushing the code to the main branch, if you are interested in supporting ogbl-vessel as well :slightly_smiling_face:. If you find anything that does not make sense or if you do think it does not make sense to merge this into main, please let me know! I am also happy to answer any questions regarding code and graph! :slightly_smiling_face: 

As `ogbl-vessel` uses the ROCAUC metric and we provide presampled negative training edges, I changed the code base of SEAL_OGB w.r.t. the following things: 

1. `seal_link_pred.py`: In addition to the rocauc function, we use a new function `evaluate_ogb_rocauc` for the ogb evaluator. I kept the old function `evaluate_auc` to support non-OGB graphs as well, as these differ slightly.
2.  `seal_link_pred.py`: Normalization of ogbl-vessel node features. 
3. `utils.py`: Use negative training samples if available.

Thank you,

Julian

### CODE REVIEW

**Comment by muhanzhang** [comment]
@jqmcginnis Hi Julian, thank you very much for the extending SEAL to support ogbl-vessel! We tested the changes and everything looks fine. We also ran it on ogbl-vessel with (1hop/5epochs/5%/use_feature/use_weight) hyperparameters and got 80.78 highest validation AUC and 80.75 final test AUC. Better results are possible with tuning. 

Cheers,\
Muhan

**Comment by jqmcginnis** [comment]
@muhanzhang Thank you very much for the integration! :slightly_smiling_face: 

Regarding submission of the results, did you do grid search for the other algorithms / previous submissions? 
As it's your algorithm and paper, please also free to report the results!
If you do not have the resources to do so, I would be happy to assist! 

Cheers, 
Julian

**Comment by muhanzhang** [comment]
Hi @jqmcginnis , we did not do grid search for this result, and mostly only did minimal hyperparameter tuning for previous SEAL submissions. I saw that in your [paper](https://arxiv.org/pdf/2108.13233.pdf) you have much higher results (90% AUC) for SEAL. Since you have better understanding of the dataset and perhaps have performed grid search for SEAL, I will appreciate it if you can submit your SEAL results to the leaderboard. Also, please feel free to refer to this repository in your ogbl-vessel official repository, and add the command to reproduce the SEAL results into the README file of this repo (or/and that of your repo). 

Thanks,\
Muhan

**Comment by jqmcginnis** [comment]
@muhanzhang 

I reported the results yesterday evening :slightly_smiling_face: - however they are not in the 90% ROC-AUC range, as the official ogbl-vessel graph differs from the VesselGraph paper in terms of node features. I will dedicate the upcoming days to clarify this in the [VesselGraph Repo](https://github.com/jocpae/VesselGraph) and we will probably update the archiv paper as well, in order to include average and std-dev for all results. 

Also, I will add ogbl-vessel to this repository's README and refer users of VesselGraph to this repo! 

Julian

File: utils.py [hunk 1]
```diff
@@ -236,13 +236,17 @@ def do_edge_split(dataset, fast_split=False, val_ratio=0.05, test_ratio=0.1):
...
+            neg_edge = split_edge[split]['edge_neg'].t()
+
+        else:
             new_edge_index, _ = add_self_loops(edge_index)
             neg_edge = negative_sampling(
                 new_edge_index, num_nodes=num_nodes,
                 num_neg_samples=pos_edge.size(1))
-        else:
-            neg_edge = split_edge[split]['edge_neg'].t()
+
```
Comment (venomouscyanide) [inline utils.py:249]: Hi @jqmcginnis and @muhanzhang ,
I was looking at this change made for the addition of `ogbl-vessel`. If I am not wrong, won't this slightly change the logic for existing datasets?
For ex., previously, if we were running the training code for `Cora`, it would enter the `if split == 'train'` block instead and create negative edges. Is there any reason we were re-sampling for negative edges previously even though this information was available already in `split_edge[split]['edge_neg']` created as part of `get_pos_neg_edges()`?

I apologize if my understanding is incorrect.
